### PR TITLE
MESH-1134/Fixes missing upgrade of substrate-rc4

### PR DESCRIPTION
**Pull request by satyamakgec**

### CODE REVIEW

File: pallets/im-online/src/lib.rs [hunk 1]
```diff
@@ -372,10 +372,11 @@ decl_module! {
...
         // block with such an extrinsic.
         #[weight = (310_000_000 + T::DbWeight::get().reads_writes(4, 1))
-            .saturating_add(750_000.saturating_mul(heartbeat.validators_len as Weight))
-            .saturating_add(
-                1_200_000.saturating_mul(heartbeat.network_state.external_addresses.len() as Weight)
-        )]
+        .saturating_add(750_000.saturating_mul(heartbeat.validators_len as Weight))
+        .saturating_add(
+            1_200_000.saturating_mul(heartbeat.network_state.external_addresses.len() as Weight)
+            )
```
Comment (maxsam4) [inline pallets/im-online/src/lib.rs:378]: indentation

File: pallets/staking/src/lib.rs [hunk 2]
```diff
@@ -2445,18 +2447,20 @@ decl_module! {
...
-            )
-            // TODO: instead of returning an error, panic. This makes the entire produced block
-            // invalid.
-            // This ensures that block authors will not ever try and submit a solution which is not
-            // an improvement, since they will lose their authoring points/rewards.
+            ).expect(
+                "An unsigned solution can only be submitted by validators; A validator should \
+                always produce correct solutions, else this block should not be imported, thus \
+				effectively depriving the validators from their authoring reward. Hence, this panic
+				is expected."
```
Comment (maxsam4) [inline pallets/staking/src/lib.rs:2461]: indentation

**Comment by maxsam4** [comment]
BTW rc5 is out already. Perhaps we should upgrade to rc5 directly?

File: pallets/staking/src/lib.rs (hunk 2 above)
Comment (satyamakgec) [reply pallets/staking/src/lib.rs:2461]: It seems fine on my local, I don't know why it seems off over github

Comment (maxsam4) [reply pallets/staking/src/lib.rs:2461]: Tab vs space issue. The bottom two lines are indented used tabs.

**Comment by satyamakgec** [comment]
When I will be done with the rc5 then they will launch the rc6 it will be a never-ending process :)

**Comment by maxsam4** [comment]
> When I will be done with the rc5 then they will launch the rc6 it will be a never-ending process :)

Yeah but since we are now committing to an evolving testnet v2 without chain-resets, we need to do these upgrades regularly.

File: pallets/staking/fuzzer/Cargo.lock [hunk 3]
```diff
@@ -0,0 +1,2294 @@
+# This file is automatically @generated by Cargo.
```
Comment (fmiguelgarcia) [inline pallets/staking/fuzzer/Cargo.lock:1]: `Cargo.lock` inside a pallet? Only root workspace should have a `Cargo.lock` file.

Comment (satyamakgec) [reply pallets/staking/fuzzer/Cargo.lock:1]: Nice catch @fmiguelgarcia I forgot about this

**Pull request merged**